ledger/sync/src/ledger_sync/ledger_sync_service.rs

```diff
@@ -188,28 +188,20 @@ impl<L: Ledger, BC: BlockchainConnection + 'static, TF: TransactionsFetcher + 's
     }
 
     /// Append safe blocks to the local ledger.
-    fn append_safe_blocks(
-        &mut self,
-        blocks_and_contents: &[(Block, BlockContents)],
-    ) -> Result<(), LedgerSyncError> {
+    fn append_safe_blocks(&mut self, blocks: &[BlockData]) -> Result<(), LedgerSyncError> {
         log::info!(
             self.logger,
             "Appending {} blocks to ledger, which currently has {} blocks",
-            blocks_and_contents.len(),
+            blocks.len(),
             self.ledger
                 .num_blocks()
                 .expect("failed getting number of blocks"),
         );
-        mc_common::trace_time!(
-            self.logger,
-            "Appended {} blocks to ledger",
-            blocks_and_contents.len()
-        );
+        mc_common::trace_time!(self.logger, "Appended {} blocks to ledger", blocks.len());
 
-        for (block, contents) in blocks_and_contents {
+        for block_data in blocks {
             let append_block_start = SystemTime::now();
-            // FIXME: Add metadata, too.
-            self.ledger.append_block(block, contents, None, None)?;
+            self.ledger.append_block_data(block_data)?;
```

Comment:
This has very nuanced and problematic behavior - it will now append the signature/timestamp of the fetched block. We don't want nodes doing catchup to publish block with a signature from another node.
We really should have tests around this...

Reply:
Good catch! I'm working on another PR to extend this properly, so I'll revert it in this PR.